### ISSUE TITLE
move full text extraction to hydra:derivatives

### ISSUE THREAD

**Issue by elrayle** [projecthydra/hydra-derivatives#90]
Move code from Hydra::Works full text extraction into hydra-derivatives allowing the use of the #make_derivatives configuration process defined by hydra-derivatives to identify that full text extraction should occur.

See issue https://github.com/projecthydra-labs/hydra-works/issues/220 in Hydra::Works for more details.

**Comment by jcoyne** [comment]
There is no `makes_derivatives` in the API any longer.

**Comment by elrayle** [comment]
Upon review, I see that a major change was made recently.  Hydra-works is not currently in sync with this change as it still uses #makes_derivatives.

This is an interesting change to the approach for generating derivatives.  Based on discussions I've heard previously, I was under the impression that there was a desire for apps to be able to choose which derivatives they wanted generated.  

For example, once full text extraction was moved over, #makes_derivatives might look something like...
```
      makes_derivatives do |obj|
        case obj.original_file.mime_type
        when *pdf_mime_types
          obj.transform_file :original_file, thumbnail: { format: 'jpg', size: '338x493' }
          obj.transform_file :original_file, ext_text: { format: 'txt' }
        when ...
        end
      end
```

If other derivatives are added and are desired for PDFs, then another transform_file call would be added for each that should be run for PDFs.  A simple example that an app might want to do now is to generate multiple thumbnails for various sizes.

The new approach seems to be deciding what gets generated for you and bundling together a predefined set of derivatives that you get based on file type.  This seems like a less flexible approach.  What were the thoughts behind making this change?

**Comment by elrayle** [comment]
BTW, the lack of #makes_derivatives doesn't effect the validity of this issue.  It does effect the implementation approach.  Also, we will have to consider whether this change goes only in the master branch destined for 3.0 or also in a new branch destined for 2.x release.

**Comment by jcoyne** [comment]
@elrayle hydra-works has been updated to use the current version of hydra-derivatives and it no longer uses makes_derivatives. See https://github.com/projecthydra-labs/hydra-works/pull/219 And yes, this is just a note to the implementer. Applications can choose which derivatives they generate by overriding create_derivatives as curation_concerns does: https://github.com/projecthydra-labs/curation_concerns/blob/master/curation_concerns-models/app/models/concerns/curation_concerns/generic_file/derivatives.rb#L14-L37

**Comment by elrayle** [comment]
I'm back up to date on the recent changes.  The original issue stands as described except that #create_derivatives block as described in the README will be used to specify which derivatives to run instead of #makes_derivatives.  Thanks Justin for the clarifications.

**Comment by jcoyne** [comment]
Closed by 51d81e75e8514f7dbdc4a1175115f3f3601e71a0
